packages/workbox-webpack-plugin/test/node/e2etest.js

```diff
@@ -0,0 +1,81 @@
+/* global workbox, expect, describe */
+const assert = require('chai').assert;
+const WorkboxBuild = require('../../../workbox-build/src');
+const proxyquire = require('proxyquire');
+const webpack = require('webpack');
+const fs = require('fs');
+const path = require('path');
+
+let WorkboxWebpackPlugin;
+
+function runWebpack(webpackConfig) {
+  return new Promise((resolve, reject) => {
+    let compiler = webpack(webpackConfig);
+    compiler.run((err, stats)=>{
+      if (err) reject(err);
+      else {
+        // Timeout for plugins that work on `after-emit` event of webpack
+        setTimeout(()=>{
+          resolve(stats);
+        }, 20);
+      }
+    });
+  });
+}
+
+describe('Tests for webpack plugin', function() {
+  this.timeout(55000);
+  beforeEach(()=>{
+    // do a proxy require
+    WorkboxWebpackPlugin = proxyquire('../../', {
+      'workbox-build': WorkboxBuild,
+    });
+  });
+
+  it('should generate sw, when `swSrc` is not present', () => {
+    const webpackConfig = {
+      entry: {
+        swApp: path.join(__dirname, 'webpack-assets/src.js'),
```

Comment:
Please change to `tmp-*` instead of a static folder.

Please add an after step to remove the directory and it's contents as well. Main advantage is that the directory should be different every test run - reduces risk of a false positive if an old test worked and a new one doesn't.